### PR TITLE
Add LifetimeDependenceScopeFixup pass

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci smoke test

File: SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+private let verbose = false
+
+private func log(_ message: @autoclosure () -> String) {
+  if verbose {
+    print("### \(message())")
+  }
+}
+
+let lifetimeDependenceScopeFixupPass = FunctionPass(
```
Comment (eeckstein) [inline SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift:24]: Can you add a comment saying what this pass actually does?

File: SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  name: "lifetime-dependence-scope-fixup")
+{ (function: Function, context: FunctionPassContext) in
+  log("Scope fixup for lifetime dependence in \(function.name)")
+
+  for instruction in function.instructions {
+    guard let markDep = instruction as? MarkDependenceInst else { continue }
```
Comment (eeckstein) [inline SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift:30]: formatting: it is more readable to put the `continue` in a separate line. Like swift-format would do.

**Review by eeckstein — APPROVED** [review]
basically lgtm.
Can you add a top-level comment describing the purpose of the pass?

File: SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift (hunk 1 above)
Comment (meg-gupta) [reply SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift:31]: Done!

**Review by atrick — APPROVED** [review]
nice!

**Pull request merged**